src/common/components/ui/OpenShiftVersionDropdown.tsx

```diff
@@ -24,32 +24,39 @@ export type HelperTextType = (
 
 type OpenShiftVersionDropdownProps = {
   name: string;
-  defaultValue: string | undefined;
   items: {
     label: string;
     value: string;
   }[];
-  isDisabled: boolean;
   versions: OpenshiftVersionOptionType[];
   getHelperText: HelperTextType;
   showReleasesLink: boolean;
 };
 
 export const OpenShiftVersionDropdown = ({
   name,
-  defaultValue,
   items,
-  isDisabled,
   versions,
   getHelperText,
   showReleasesLink,
 }: OpenShiftVersionDropdownProps) => {
   const [field, , { setValue }] = useField(name);
   const [isOpen, setOpen] = React.useState(false);
-  const [current, setCurrent] = React.useState(defaultValue);
   const { t } = useTranslation();
-  const [helperText, setHelperText] = React.useState(getHelperText(versions, defaultValue, t));
   const fieldId = getFieldId(name, 'input');
+  const isDisabled = versions.length === 0;
+
+  const { defaultLabel, defaultValue } = React.useMemo(() => {
+    const defaultVersion = versions.find((item) => item.default);
+    return {
+      defaultLabel: defaultVersion?.label || '',
+      defaultValue: defaultVersion?.value || '',
+    };
+  }, [versions]);
+
+  const [helperText, setHelperText] = React.useState(getHelperText(versions, defaultValue, t));
+  const [current, setCurrent] = React.useState<string>(defaultLabel);
```

Comment:
`current` what? ;-P

Reply:
Agree, it should be `currentLabel` but it's like that in possibly all Dropdowns and I would rename it for all at a later point.